Add common conversion traits for Node types.

Conner-PYS opened the pull request:
Thanks for putting this library together! I've been testing it out over the last day or so and found myself wanting easier from/into conversions for some of the Node types and strings.

This PR adds the following impls:
* on `Node`:
  * `From<Comment>`
  * `From<Doctype>`
  * `From<Fragment>`
  * `From<Element>`
  * `From<Text>`
  * `From<UnsafeText>`
* on `Comment`:
  * `From<String>`
  * `From<&str>`
* on `Doctype`:
  * `From<String>`
  * `From<&str>`
* on `Text`:
  * `From<String>`
  * `From<&str>`
* on `UnsafeText`:
  * `From<String>`
  * `From<&str>`

vidhanio commented on html-node-core/src/node/unsafe_text.rs:
```diff
@@ -0,0 +1,34 @@
+}
+
+impl Display for UnsafeText {
+    /// Unformatted text.
```
nit: "unescaped" may be better?

Conner-PYS replied:
updated!

vidhanio commented on html-node/Cargo.toml:
style: would be appreciated if you could revert the formatting applied here, reducing the amount of diff coverage needed.

Conner-PYS replied:
Yep, fixed! Sorry, my editor had TOML auto-formatting turned on; works great for my own projects, but just makes a mess other times.

vidhanio commented on html-node-core/src/node/doctype.rs:
````diff
@@ -0,0 +1,37 @@
+    /// The value of the doctype.
+    ///
+    /// ```html
+    /// <!DOCTYPE synax>
````
(this is my typo) but it would be appreciated if you could fix this to `The syntax of the doctype.` and change `synax` to `syntax` in the codeblock because i don't wanna create merge conflicts :)

Conner-PYS replied:
fixed

vidhanio reviewed (changes requested):
just a few notes, but these look extremely helpful for making the api easier to use, thank you very much for your contributions!

i apologize for not recognizing this pr sooner, i completely missed it in my github notifications.

Conner-PYS commented:
Ok, I've added a new commit with the final update I need at the moment, which is better support for inserting CSS content. Check out the new `style!()` proc-macro and `styling.rs` example I added to the main crate. This macro is feature-flagged behind `basic-css`.

I've tried to make sure that all the relevant documentation has also been updated, but let me know if something isn't clear.

Version has been bumped to `0.3` so you can publish to crates.io without any further commits.

And no worries about missing the PR for a bit, we've all got lives!

vidhanio commented:
@Conner-PYS, would it be possible to move css changes to a new pr? there is some further discussion that would be had there, which I don't want blocking the conversion changes.

these css changes look great btw!

Conner-PYS commented:
Pulled the CSS macro out; I'll put up another PR with it soon. I'd love to hear if you have any ideas on improving the ergonomics of writing CSS in Rust tokens.

Conner-PYS commented:
CSS macro draft PR is here: https://github.com/vidhanio/html-node/pull/3

Simply reverts the last commit from this PR at the moment.
